### PR TITLE
SALTO-2748: Add e2e tests for Okta

### PR DESCRIPTION
**Pull request by shir-reifenberg**
e2e tests for Okta

---

Tests:

1. All types created during fetch
2. build-in instances returns on fetch
3. deploy of `Group` + `GroupRule` with template expression

Still TODO:

- [X]  Test for deployment of `Application`, `ProfileEnrollment`, `ProfileEnrollmentPolicyRule` - will add later
- [X] Add e2e to private api types - opened a ticket for now

---

### CODE REVIEW

File: packages/okta-adapter/e2e_test/adapter.test.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import { CredsLease } from '@salto-io/e2e-credentials-store'
+import _ from 'lodash'
+import { Credentials } from '../src/auth'
+import { credsLease, realAdapter } from './adapter'
+import 'jest-extended'
+import OktaAdapter from '../src/adapter'
+import { DEFAULT_CONFIG } from '../src/config'
+import { GROUP_TYPE_NAME, APPLICATION_TYPE_NAME } from '../src/constants'
+
+jest.setTimeout(300 * 1000)
```
Comment (netama) [inline packages/okta-adapter/e2e_test/adapter.test.ts:27]: does it take that long? or copied from somewhere else?

File: packages/okta-adapter/e2e_test/adapter.test.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+describe('Okta E2E', () => {
+  let fetchedElements: Element[]
+  let credLease: CredsLease<Credentials>
+  let adapter: OktaAdapter
+
+  beforeAll(async () => {
+    credLease = await credsLease()
+    const config = _.cloneDeep(DEFAULT_CONFIG);
+    (config.apiDefinitions.types.Application.transformation as configUtils.TransformationConfig).idFields = ['name', 'status']
```
Comment (netama) [inline packages/okta-adapter/e2e_test/adapter.test.ts:37]: why did we need to override this? isn't this the default?

File: packages/okta-adapter/jest.config.js [hunk 3]
```diff
@@ -21,15 +21,18 @@ module.exports = deepMerge(
...
     collectCoverageFrom: [
       '!<rootDir>/index.ts',
     ],
     coverageThreshold: {
       'global': {
-        branches: 32,
-        functions: 52,
-        lines: 65,
-        statements: 66,
+        branches: 10,
```
Comment (netama) [inline packages/okta-adapter/jest.config.js:32]: hmm did this drop because of the added tests?

**Review by netama — COMMENTED** [review]
looks good, added some questions

File: packages/okta-adapter/src/config.ts [hunk 4]
```diff
@@ -440,7 +440,7 @@ const DEFAULT_TYPE_CUSTOMIZATIONS: OktaSwaggerApiConfig['types'] = {
         { fieldName: 'scheme', fieldType: 'string' },
         { fieldName: 'userName', fieldType: 'string' },
       ],
-      fieldsToHide: [{ fieldName: 'signing', fieldType: 'ApplicationCredentialsSigning' }],
+      fieldsToOmit: [{ fieldName: 'signing', fieldType: 'ApplicationCredentialsSigning' }],
```
Comment (alonstern) [inline packages/okta-adapter/src/config.ts:443]: Is this change related to the PR?

**Review by alonstern — COMMENTED** [review]
Looks good 👌

File: packages/okta-adapter/src/config.ts (hunk 4 above)
Comment (shir-reifenberg) [reply packages/okta-adapter/src/config.ts:443]: It should be in `fieldsToOmit`, I just noticed it now cause the e2e failed for this

**Review by alonstern — APPROVED** [review]
